### PR TITLE
Add RootGauge contracts

### PR DESCRIPTION
**Pull request by hannespfeiffer**
Add 'RootGaugeArbitrum'
 - adopted from 'RootGaugeLocal.vy'
 - added 'name' field for frontend. LP token name of associated pool on L2 should be given as constructor argument

### CODE REVIEW

**Comment by penandlim** [comment]
Could you also add name field to RootGaugeLocal as well? Looks like frontend folks are encountering issues on hardhat node due to it.

File: contracts/tokenomics/gauges/sidechain/RootGaugeEvmos.vy [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    gas_price: uint256
+    max_submission_cost: uint256
+
+
+## BRIDGE AND L2 SPECIFIC CONSTANTS 
+
+BRIDGE_ROUTER: constant(address) = 0x88A69B4E698A4B090DF6CF5Bd7B2D47325Ad30A3
+NOMAD_EVMOS_MAINNET_DESTINATION_CODE: constant(uint32) = 1702260083
+NOMAD_BRIDGE_ENABLE_FAST_TRANSFER: constant(bool) = False 
+NAME: immutable(String[64])
```
Comment (penandlim) [inline contracts/tokenomics/gauges/sidechain/RootGaugeEvmos.vy:49]: Along with the immutable constant `NAME`, we also need to add a consistent name function/variable with ERC20 ABI.

In this case that would be a function with lowercase like below.

```vyper
@view
@external
def name() -> String[64]:
  """
  @notice Get the name for this gauge token
  """
  return NAME
```

File: contracts/tokenomics/gauges/sidechain/RootGaugeArbitrum.vy [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+event FeesModified:
+    gas_limit: uint256
+    gas_price: uint256
+    max_submission_cost: uint256
+
+
+## BRIDGE AND L2 POOL SPECIFIC CONSTANTS 
+GATEWAY_ROUTER: constant(address) = 0x72Ce9c846789fdB6fC1f34aC4AD25Dd9ef7031ef
+GATEWAY: constant(address) = 0xa3A7B6F88361F48403514059F1F16C8E78d60EeC
+NAME: immutable(String[64])
```
Comment (penandlim) [inline contracts/tokenomics/gauges/sidechain/RootGaugeArbitrum.vy:47]: See Comment below

File: contracts/tokenomics/gauges/sidechain/RootGaugeOptimism.vy [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    gas_price: uint256
+    max_submission_cost: uint256
+
+
+## BRIDGE AND L2 SPECIFIC CONSTANTS 
+
+L1_STANDARD_BRIDGE: constant(address) = 0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1
+L2_SDL_TOKEN: constant(address) = 0xAe31207aC34423C41576Ff59BFB4E036150f9cF7
+L2_GAS: constant(uint32) = 200_000
+NAME: immutable(String[64])
```
Comment (penandlim) [inline contracts/tokenomics/gauges/sidechain/RootGaugeOptimism.vy:49]: Ditto here

**Review by penandlim — CHANGES_REQUESTED** [review]

File: contracts/tokenomics/gauges/sidechain/RootGaugeEvmos.vy (hunk 1 above)
Comment (hannespfeiffer) [reply contracts/tokenomics/gauges/sidechain/RootGaugeEvmos.vy:49]: Would we? The rootgauges don't implement the ERC20 interface. Or does frontend need a dedicated getter function?

**Review by penandlim — APPROVED** [review]

**Pull request merged**